contracts/FiatToken.cdc

```diff
@@ -1,1060 +1,922 @@
+	
+	access(all)	event ControllerConfigured(controller: UInt64, minter: UInt64)
+	access(all)	event ControllerRemoved(controller: UInt64)
+
+	// ------- FungibleToken Events -------
+	access(all)	event TokensWithdrawn(amount: UFix64, from: Address?)
+	access(all)	event TokensDeposited(amount: UFix64, to: Address?)
+	
+	// ------- FiatToken Paths -------
+	access(all)	let VaultStoragePath: StoragePath
+	access(all)	let VaultBalancePubPath: PublicPath
+	access(all)	let VaultUUIDPubPath: PublicPath
+	access(all)	let VaultReceiverPubPath: PublicPath
+	access(all)	let BlocklistExecutorStoragePath: StoragePath
+	access(all)	let BlocklisterStoragePath: StoragePath
+	access(all)	let BlocklisterCapReceiverPubPath: PublicPath
+	access(all)	let BlocklisterUUIDPubPath: PublicPath
+	access(all)	let BlocklisterPubSigner: PublicPath
+	access(all)	let PauseExecutorStoragePath: StoragePath
+	access(all)	let PauserStoragePath: StoragePath
+	access(all)	let PauserCapReceiverPubPath: PublicPath
+	access(all)	let PauserUUIDPubPath: PublicPath
+	access(all)	let PauserPubSigner: PublicPath
+	access(all)	let AdminExecutorStoragePath: StoragePath
+	access(all)	let AdminStoragePath: StoragePath
+	access(all)	let AdminCapReceiverPubPath: PublicPath
+	access(all)	let AdminUUIDPubPath: PublicPath
+	access(all)	let AdminPubSigner: PublicPath
+	access(all)	let OwnerExecutorStoragePath: StoragePath
+	access(all)	let OwnerStoragePath: StoragePath
+	access(all)	let OwnerCapReceiverPubPath: PublicPath
+	access(all)	let OwnerUUIDPubPath: PublicPath
+	access(all)	let OwnerPubSigner: PublicPath
+	access(all)	let MasterMinterExecutorStoragePath: StoragePath
+	access(all)	let MasterMinterStoragePath: StoragePath
+	access(all)	let MasterMinterCapReceiverPubPath: PublicPath
+	access(all)	let MasterMinterUUIDPubPath: PublicPath
+	access(all)	let MasterMinterPubSigner: PublicPath
+	access(all)	let MinterControllerStoragePath: StoragePath
+	access(all)	let MinterControllerUUIDPubPath: PublicPath
+	access(all)	let MinterControllerPubSigner: PublicPath
+	access(all)	let MinterStoragePath: StoragePath
+	access(all)	let MinterUUIDPubPath: PublicPath
+	
+	// ------- FiatToken States / Variables -------
+	access(all)	let name: String
+	
+	access(all)	var version: String
+	
+	// Set to true if the contract is paused
+	access(all)	var paused: Bool
+	
+	// The token total supply
+	access(all)	var totalSupply: UFix64
+	
+	// Blocked resources dictionary {resourceId: Block Height}
+	access(contract) let blocklist: {UInt64: UInt64}
+	
+	// Managed minters dictionary {MinterController: Minter}
+	access(contract) let managedMinters: {UInt64: UInt64}
+	
+	// Minter allowance dictionary {Minter: Allowance}
+	access(contract) let minterAllowances: {UInt64: UFix64}
+	
+	// ------- FiatToken Interfaces  -------
+	access(all)	resource interface ResourceId { 
+		access(all) view fun UUID(): UInt64 {
             return self.uuid
         }
+	}
+
```

Comment:
I added this interface so we could include default implementations for all of the `PublicSigner` functions and remove a lot of the boilerplate code.

This also required having the `multiSigManager` field be `access(Owner)` instead of `access(self)`, which is something that needs to be considered